File: src/main/java/com/kamikazejam/datastore/base/log/CacheLoggerService.java
package com.kamikazejam.datastore.base.log;

import com.kamikazejam.datastore.DataStoreAPI;
import com.kamikazejam.datastore.base.Cache;
import org.bukkit.plugin.Plugin;

public class CacheLoggerService extends LoggerService {

    protected final Cache<?, ?> cache;

    public CacheLoggerService(Cache<?, ?> cache) {
        this.cache = cache;
    }

    @Override
    public boolean isDebug() {
        return DataStoreAPI.isDebug();
    }

    @Override
    public String getLoggerName() {
        return "C: " + cache.getName();
    }

    @Override
    public Plugin getPlugin() {
        return cache.getPlugin();
    }
}

File: src/main/kotlin/com/kamikazejam/datastore/base/log/CacheLoggerService.kt
package com.kamikazejam.datastore.base.log

import com.kamikazejam.datastore.DataStoreAPI
import com.kamikazejam.datastore.base.Cache
import org.bukkit.plugin.Plugin

open class CacheLoggerService(protected val cache: Cache<*, *>) : LoggerService() {
    override val isDebug: Boolean
        get() = DataStoreAPI.isDebug()

    override val loggerName: String
        get() = "C: " + cache.name

    override val plugin: Plugin
        get() = cache.plugin
}
